tgui/packages/tgui/interfaces/CentcomPodLauncher.js

```diff
@@ -1,518 +1,1125 @@
-              selected={data.styleChoice === 11}
-              tooltip="For when an orange is angry"
-              onClick={() => act('styleFruit')} />
-            <Button
-              content="Invisible"
-              selected={data.styleChoice === 12}
-              tooltip={multiline`
-                Makes the supplypod invisible! Useful for when you want to
-                use this feature with a gateway or something. Combos well
-                with the "Stealth" and "Quiet Landing" effects.
-              `}
-              onClick={() => act('styleInvisible')} />
-            <Button
-              content="Gondola"
-              selected={data.styleChoice === 13}
-              tooltip={multiline`
-                This gondola can control when he wants to deliver his supplies
-                if he has a smart enough mind, so offer up his body to ghosts
-                for maximum enjoyment. (Make sure to turn off bluespace and
-                set an arbitrarily high open-time if you do!
-              `}
-              onClick={() => act('styleGondola')} />
+                selected={
+                  option.key
+                    ? data.reverseOptionList[option.key]
+                    : data.reverseOptionList[option.title]
+                }
+                tooltip={option.title}
+                tooltipOverrideLong
+                onClick={() => act('reverseOption', {
+                  reverseOption: option.key
+                    ? option.key
+                    : option.title })} />
+            ))}
+          </Flex.Item>
+        </Flex>
+      )}
+    </Section>
+  );
+};
+
+const saveDataToPreset = (id, data) => {
+  storage.set("podlauncher_preset_"+id, data);
+};
+
+const loadDataFromPreset = (id, context) => {
+  const { act, data } = useBackend(context);
+  act('loadDataFromPreset', { payload: storage.get("podlauncher_preset_"+id) });
+};
+
+const newPreset = (presetName, hue, data) => {
+  const current = getPresets();
+  let id = CreateUuid();
+  current.push({ id: id, title: presetName, hue: hue });
+  storage.set("podlauncher_presetlist", current);
+  saveDataToPreset(id, data);
+};
+
+const getPresets = id => {
+  return storage.get("podlauncher_presetlist") || [];
+};
+
+const deletePreset = deleteID => {
+  const current = getPresets();
+  for (let i = 0; i < current.length; i++) {
+    if (current[i].id === deleteID) {
+      current.splice(i, 1);
+    }
+  }
+  storage.set("podlauncher_presetlist", current);
+};
+
+const PresetsPage = (props, context) => {
+  const { act, data } = useBackend(context);
+  const [
+    presetIndex,
+    setSelectedPreset,
+  ] = useLocalState(context, 'presetIndex', 0);
+  const [
+    settingName,
+    setEditingNameStatus,
+  ] = useLocalState(context, 'settingName', 0);
+  const [newNameText, setText] = useLocalState(context, 'newNameText', "");
+  const [hue, setHue] = useLocalState(context, 'hue', 0);
+  const presets = getPresets();
```

Comment:
Ah well, nope, I was wrong, this will not cleanly migrate to async.

You would need unfortunately to convert this component into a class-based component, and then initiate retrieval of presets when component mounts.

This also has the benefit of not spamming storage calls, which are pretty slow and expensive.

```js
class PresetsPage extends Component {
  constructor() {
    super();
    this.state = {
      presets: [],
    };
  }

  async componentDidMount() {
    this.setState({
      presets: await getPresets(),
    });
  }

  render() {
    const { props, context } = this;
    const { presets } = this.state;
    ...
  }
}
```